sklearn/utils/estimator_checks.py

```diff
@@ -3126,12 +3126,17 @@ def check_n_features_in_after_fitting(name, estimator_orig):
     X = rng.normal(size=(n_samples, 8))
     X = _enforce_estimator_tags_x(estimator, X)
     X = _pairwise_estimator_convert_X(X, estimator)
+
     if is_regressor(estimator):
         y = rng.normal(size=n_samples)
     else:
         y = rng.randint(low=0, high=2, size=n_samples)
     y = _enforce_estimator_tags_y(estimator, y)
 
+    if name in CROSS_DECOMPOSITION:
```

Comment:
could you please add a comment as why this is needed for future reference?

Reply:
is it making sense to have this in `_enforce_estimator_tags_y` with proper tags?

Reply:
As a quick fix, we acn put it into `_enforce_estimator_tags_y` first without a tag. We do this `name in CROSS_DECOMPOSITION` "hack" a bunch in `estimator_checks`, it would be worth it to centralize.

An alternative would be to `n_components=1` in `_set_checking_parameters` for the `cross_decomposition`. I went with this option for this PR.